# named list of all robots
fileversion: 3

robots:
  # cf231:
  #   enabled: true
  #   uri: radio://0/80/2M/E7E7E7E7E7
  #   initial_position: [0.0, 0.0, 0.0]
  #   type: cf21  # see robot_types
  cf231:
    enabled: true
    uri: radio://0/88/2M/E7E7E7EE07
    initial_position: [0.0, 0.0, 0.0]
    type: cf21  # see robot_types
    firmware_params:
      # the flapper had some strange bias in position
      # fix it by using an integral gain
      posCtlPid:
        xKi: 0.5
        yKi: 0.5

  cf5:
    enabled: false
    uri: radio://0/80/2M/E7E7E7EE02
    initial_position: [0.0, -0.2, 0.0]
    type: cf21  # see robot_types

    # firmware_params:
    #   kalman:
    #     pNAcc_xy: 1.0 # default 0.5
    # firmware_logging:
    #   enabled: true
    #   custom_topics:
    #     topic_name3: 
    #       frequency: 1
    #       vars: ["acc.x", "acc.y"]
    # reference_frame: "odom_cf231" # use a custom reference frame if no global coordinate system is available

# Definition of the various robot types
robot_types:
  cf21:
    motion_capture:
      tracking: "librigidbodytracker" # one of "vendor", "librigidbodytracker"
      # only if enabled; see motion_capture.yaml
      marker: default_single_marker
      dynamics: default
    big_quad: false
    battery:
      voltage_warning: 3.8  # V
      voltage_critical: 3.7 # V
    # firmware_params:
    #   kalman:
    #     pNAcc_xy: 1.0 # default 0.5
    #firmware_logging:
    #   enabled: true
    #   default_topics:
    #   pose:
    #     frequency: 1 # Hz
    #   custom_topics:
    #    topic_name3: 
    #      frequency: 1
    #      vars: ["acc.x", "acc.y", "acc.z"]
    # reference_frame: "odom" 

  cf21_mocap_deck:
    motion_capture:
      tracking: "librigidbodytracker" # one of "vendor", "librigidbodytracker"
      # only if enabled; see motion_capture.yaml
      marker: mocap_deck
      dynamics: default
    big_quad: false
    battery:
      voltage_warning: 3.8  # V
      voltage_critical: 3.7 # V
    # firmware_params:
    #   kalman:
    #     pNAcc_xy: 1.0 # default 0.5
    #firmware_logging:
    #   enabled: true
    #   default_topics:
    #   pose:
    #     frequency: 1 # Hz
    #   custom_topics:
    #    topic_name3: 
    #      frequency: 1
    #      vars: ["acc.x", "acc.y", "acc.z"]
    # reference_frame: "mocap" 

    # Just a marker to track, not a robot to connect to
  pololu:
    connection: none
    motion_capture:
      tracking: "vendor"


# global settings for all robots
all:
  # firmware logging for all drones (use robot_types/type_name to set per type, or
  # robots/drone_name to set per drone)
  firmware_logging:
    enabled: true
    default_topics:
     # remove to disable default topic
      # pose:
      #   frequency: 10 # Hz
      # status:
      #   frequency: 1 # Hz

      odom:
        frequency: 10 # Hz
      scan:
        frequency: 10 # Hz

    #custom_topics:
    #  topic_name1:
    #    frequency: 10 # Hz
    #    vars: ["stateEstimateZ.x", "stateEstimateZ.y", "stateEstimateZ.z", "pm.vbat"]
    #  topic_name2:
    #    frequency: 1 # Hz
    #    vars: ["stabilizer.roll", "stabilizer.pitch", "stabilizer.yaw"]
  # firmware parameters for all drones (use robot_types/type_name to set per type, or
  # robots/drone_name to set per drone)
  firmware_params:
    commander:
      enHighLevel: 1
    stabilizer:
      estimator: 2 # 1: complementary, 2: kalman
      controller: 1 # 1: PID, 2: mellinger
    # ring:
    #   effect: 16 # 6: double spinner, 7: solid color, 16: packetRate
    #   solidBlue: 255 # if set to solid color
    #   solidGreen: 0 # if set to solid color
    #   solidRed: 0 # if set to solid color
    #   headlightEnable: 0
    locSrv:
      extPosStdDev: 1e-3
      extQuatStdDev: 0.5e-1
    # kalman:
    #   resetEstimation: 1
  # reference frame for on-board state-estimate of all drones (use robot_types/type_name to set per type, or
  # robots/drone_name to set per drone)
  # reference_frame: "world"
  reference_frame: "cf231/odom"

  broadcasts:
    num_repeats: 15 # number of times broadcast commands are repeated
    delay_between_repeats_ms: 1 # delay in milliseconds between individual repeats
